sys/vm/vm_kern.c

```diff
@@ -246,6 +246,7 @@ kmem_alloc_attr_domain(int domain, vm_size_t size, int flags, vm_paddr_t low,
 			pmap_zero_page(m);
 		vm_page_valid(m);
 		VM_OBJECT_ASSERT_CAP(object, prot);
+		vm_page_aflag_set(m, PGA_CAPSTORE | PGA_CAPDIRTY);
```

Comment:
Should this be conditional on `prot & VM_PROT_WRITE_CAP`?  This would make the code DTRT if we choose to add an 'M_CAP' in the future to permit allocation of kernel pages that can't hold caps.

Reply:
Seems sensible enough; done.